driver/src/price_estimation/dexag.rs

```diff
@@ -177,38 +183,39 @@ mod tests {
             Token::new(4, "USDC", 6),
         ];
 
-        let api_tokens = [
-            super::api::Token {
-                name: String::new(),
-                symbol: "DAI".to_string(),
-                address: None,
-            },
-            super::api::Token {
-                name: String::new(),
-                symbol: "ETH".to_string(),
-                address: None,
-            },
-            super::api::Token {
-                name: String::new(),
-                symbol: "USDC".to_string(),
-                address: None,
-            },
-        ];
+        lazy_static! {
+            static ref API_TOKENS: [super::api::Token; 3] = [
```

Comment:
nit: This doesn't have to be `lazy_static` right? Is this because of lifetime issues with the boxed futures? I feel like scoping the `MockDexagApi` to a smaller lifetime and then using a borrow *should* work (but I may be wrong).

Shouldn't `Rc` (or `Arc` if it needs to be `Send`) also work here?

Reply:
This is about lines like `returning(|| async { Ok(API_TOKENS.to_vec()) }.boxed())`.
This closure's lifetime has to be 'static because of a mockall limitiation. Previously this was accomplished by cloning the api tokens and moving into the closure. This is still possible with the move to async but I find this way cleaner. Otherwise it looks like
```rust
        let a = api_tokens.clone();
        api.expect_get_token_list().returning(move || {
            let a = a.clone();
            async move { Ok(a.to_vec()) }.boxed()
        });
```

Reply:
Yuck! `lazy_static` makes a lot of sense then. Thanks for clarifying!